### ISSUE TITLE
The presence of two or more same-name html elements inside another html element will produce borked output

### ISSUE THREAD

**Issue by bathos** [probablyup/markdown-to-jsx#125]
I couldn’t determine precisely what the trigger is — I’m not certain it’s really specific to these element names — but strange results can be reliably produced with dl, dt or dd elements and seemingly not with other made-up two-letter element names, e.g. "aa", "bb", "cc".

```
> console.log(JSON.stringify(require('markdown-to-jsx').compiler('<dl><dt>foo</dt><dd>bar</dd><dt>baz</dt><dd>qux</dd></dl>'), undefined, 2))
{
  "type": "dl",
  "key": "0",
  "ref": null,
  "props": {
    "children": [
      {
        "type": "dt",
        "key": "0",
        "ref": null,
        "props": {
          "children": [
            "foo",
            {
              "type": "/dt><dd>bar</dd><dt",
              "key": "1",
              "ref": null,
              "props": {},
              "_owner": null,
              "_store": {}
            },
            "baz"
          ]
        },
        "_owner": null,
        "_store": {}
      },
      {
        "type": "dd",
        "key": "1",
        "ref": null,
        "props": {
          "children": [
            "qux"
          ]
        },
        "_owner": null,
        "_store": {}
      }
    ]
  },
  "_owner": null,
  "_store": {}
}
```

**Comment by bathos** [comment]
Found it. The HTML_BLOCK_ELEMENT_R pattern is greedy about its content:

    const HTML_BLOCK_ELEMENT_R = /^ *<([^ >]*) ?([^>]*)>((?:.|(?!\n *<\1)\n)*)<\/\1>\n*/;

This can be (sorta) fixed by changing it to:

    const HTML_BLOCK_ELEMENT_R = /^ *<([^ >]*) ?([^>]*)>((?:.|(?!\n *<\1)\n)*?)<\/\1>\n*/;

But looking at this pattern, I’m pretty sure there are a bunch of other ways it can go south even with that fix though. This may just not be a good way to capture HTML.

**Comment by probablyup** [comment]
> But looking at this pattern, I’m pretty sure there are a bunch of other ways it can go south even with that fix though. This may just not be a good way to capture HTML.

That's definitely true, but I'm sort of okay with that for the time being since this strategy strikes a good balance between functionality and bundle size. Markdown in HTML tends to be rare, so we'll fix the issues as they come up and I'll work on putting together more comprehensive test cases.

**Comment by probablyup** [comment]
Released in 6.1.4
